fc/protocol.py

```diff
@@ -241,9 +239,31 @@ def _parse(self):
         self.input_env.execute_statements(self.inputs)
 
         # Store unit definitions and add these to protocol
-        self.unit_definitions = details.get('units', [])
-        for units in self.unit_definitions:
-            self.units.add_unit(units.name, units.pint_expression)
+        self.unit_definitions = {udef.name: udef for udef in details.get('units', [])}
```

Comment:
Would it make more sense to just to `self.unit_definitions = {}` and then iterate only once?